integration/connector_test.go

```diff
@@ -0,0 +1,317 @@
+// Copyright (c) 2022 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+
+//go:build integration
+
+package integration
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	env "github.com/caarlos0/env/v6"
+
+	"github.com/eclipse-kanto/kanto/integration/util"
+	"github.com/eclipse/ditto-clients-golang"
+	"github.com/eclipse/ditto-clients-golang/model"
+	"github.com/eclipse/ditto-clients-golang/protocol"
+	"github.com/eclipse/ditto-clients-golang/protocol/things"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"golang.org/x/net/websocket"
+)
+
+type suiteConnectorTestConfig struct {
+	StatusTimeoutMs int `env:"SCT_STATUS_TIMEOUT_MS" envDefault:"10000"`
+
+	TimeDeltaMs int `env:"SCT_TIME_DELTA_MS" envDefault:"0"`
+	TimeSleepMs int `env:"SCT_TIME_SLEEP_MS" envDefault:"2000"`
+}
+
+type ConnectorSuite struct {
+	suite.Suite
+	suiteInitializer *util.SuiteInitializer
```

Comment:
Use golang struct embedding and don't use pointers here. All members of SuiteInitializer are pointers deliberately to make it possible to use this without external pointers.